### ISSUE TITLE
Background elements of Dialog/Drawer should be unfocus-able and hidden. Headers in background should not show up in screenreader heaver navigation

### ISSUE THREAD

**Issue by faye-tan** [palantir/blueprint#6756]
#### Environment

- __Package version(s)__: 5.8.2
- __Browser and OS versions__: Edge, Chrome, MacOS Sonoma 14.4 

#### Feature request

When a Dialog or Drawer is open, the background elements should be un-focusable and hidden. This is backed up by this [source](https://accessibilityinsights.io/info-examples/web/aria-hidden-body/) and this [source](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/dialog/). Otherwise, the screenreader cursor could focus on something behind the Dialog/Drawer. 

#### Examples

Currently when navigating by headings (CTRL+OPT+U on Mac) using a screenreader, currently when a Dialog or Drawer is open, the headers in the content underneath the overlay are also included in header navigation, which is unexpected.
